core/templates/dev/head/services/LocalStorageService.js

```diff
@@ -0,0 +1,105 @@
+// Copyright 2017 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Utility service for saving data locally on the client machine.
+ */
+
+ // Service for saving exploration draft changes to local storage.
+ //
+ // Note that the draft is only saved if localStorage exists and works
+ // (i.e. has storage capacity).
+ oppia.factory('LocalStorageService', [
+   'ExplorationDraftObjectFactory',
+   function(ExplorationDraftObjectFactory) {
+     // Check that local storage exists and works as expected.
+     // If it does storage stores the localStorage object,
+     // else storage is undefined or false.
+     var storage = (function() {
+       var test = 'test';
+       var result;
+       try {
+         localStorage.setItem(test, test);
+         result = localStorage.getItem(test) === test;
+         localStorage.removeItem(test);
+         return result && localStorage;
+       } catch (exception) {}
+     }());
+     /**
+     * Create the key to access the changeList in localStorage
+     * @param {String} explorationId - The exploration id of the changeList
+     *   to be accessed.
+     */
+     var _createExplorationDraftKey = function(explorationId) {
+       return 'draft_' + explorationId;
+     }
+     return {
+       /**
+       * Check that localStorage is available to the client.
+       * @returns {boolean} true iff the client has access to localStorage.
+       */
+       isStorageAvailable: function() {
+         return Boolean(storage);
+       },
+       /**
+       * Save the given changeList to localStorage along with its
+       * draftChangeListId
+       * @param {String} explorationId - The id of the exploration
+       *   associated with the changeList to be saved.
+       * @param {List} changeList - The exploration change list to be saved.
+       * @param {Integer} draftChangeListId - The id of the draft to be saved.
+       */
+       saveExplorationDraft: function(explorationId, changeList,
```

Comment:
When contents of parens span more than one line, break after '('. You can indent the following line by an additional two spaces, too, to distinguish its left edge from the stuff that's actually inside the (...).

Reply:
Done.